OpenRA.Mods.Common/Activities/LayMines.cs

```diff
@@ -60,6 +61,12 @@ public override bool Tick(Actor self)
 		{
 			returnToBase = false;
 
+			if (layingMine)
+			{
+				layingMine = false;
+				FinishLayingMine(self);
+			}
```

Comment:
now when a delay gets added, the you don't exit the code

Reply:
So maybe something like this:

```suggestion
			{
				layingMine = false;
				if (!FinishLayingMine(self))
				{
					QueueChild(new Wait(minelayer.Info.AfterLayingDelay));
					return false;
				}
			}
```

and this:

```
// in Tick()
if (minelayer.Info.PreLayDelay == 0)
{
	if (!FinishLayingMine(self))
		QueueChild(new Wait(minelayer.Info.AfterLayingDelay));
}
else
{
	layingMine = true;
	QueueChild(new Wait(minelayer.Info.PreLayDelay, false));
}
```

```
bool FinishLayingMine(Actor self)
{
	minefield.Remove(self.Location);

	// ...

	return minelayer.Info.AfterLayingDelay == 0;
}
```

i.e. what you suggested before?

Reply:
Something like that, just IMO it would be better to move the `minelayer.Info.AfterLayingDelay == 0` condition into the Tick loop, as from the function name it's not clear what the bool means. Probably best to keep the `FinishLayingMine` a void.

It might also be better to rename `FinishLayingMine` to just `LayMine`. Out of scope but it might also be good to rename `CanLayMine` to `IsPositionSuitableForMine` or smth along those lines.